### PR TITLE
ECO-48: Add Data Union chapter to README

### PR DESCRIPTION
**Pull request by kare**

### CODE REVIEW

**Comment by hpihkala** [comment]
@isentropy @fonty1 one of the last loose ends - could you guys review this please?

**Comment by hpihkala** [comment]
On a quick glance, looks like the `sendToMainnet` argument added in #148 is missing from the method signatures. @kare could you check that this is up to date?

**Comment by isentropy** [comment]
> @isentropy @fonty1 one of the last loose ends - could you guys review this please?

yes, working on docs. I'll commit today

File: README.md [hunk 1]
```diff
@@ -361,6 +362,70 @@ To stop receiving events from a stream, pass the `Subscription` object you got w
...
 ```
 
+<a name="data-unions"></a>
+## Data Unions
+This library provides functions for working with Data Unions. To get a DataUnion client instance, call
+`client.dataUnionClient(mainnetAdminPrivateKey, sideChainAdminPrivateKey)`. To deploy a new DataUnion, call
+`dataUnionClient.deployDataUnion(name, adminAddress, adminFeeFractionWei, agents)`. To get an existing instance of
+Data Union call `dataUnionClient.dataUnionFromName(name)`.
+
+### Functions
```
Comment (fonty1) [inline README.md:372]:
```suggestion
### Admin functions
```

File: README.md [hunk 2]
```diff
@@ -361,6 +362,70 @@ To stop receiving events from a stream, pass the `Subscription` object you got w
...
+
+| Name          | Returns  | Description |
+| :------------ | :------ | :----------- |
+| isDeployed() | boolean | Check if Data Union deployment has completed |
+| addMembers(String ...members) | Transaction receipt | Add members |
+| partMembers(String ...members) | Transaction receipt | Remove members from Data Union |
+| withdrawTokensForSelfOrAsAdmin(String memberAddress, BigInteger amount) | Transaction receipt | Withdraw members tokens to given address |
+| withdrawTokensForMember(BigInteger privateKey, String to, BigInteger amount) | Transaction receipt | Withdraw members tokens |
+
+Here's an example how to deploy a data union contract and set the admin fee:
```
Comment (fonty1) [inline README.md:382]:
```suggestion
Here's an example on how to deploy a Data Union contract and set the admin fee:
```

File: README.md [hunk 3]
```diff
@@ -361,6 +362,70 @@ To stop receiving events from a stream, pass the `Subscription` object you got w
 client.unsubscribe(sub);
 ```
 
+<a name="data-unions"></a>
+## Data Unions
+This library provides functions for working with Data Unions. To get a DataUnion client instance, call
```
Comment (fonty1) [inline README.md:367]: @kare can you spread these out, like 

This library provides functions for working with Data Unions. 

To get a DataUnion client instance, call:
```
client.dataUnionClient(mainnetAdminPrivateKey, sideChainAdminPrivateKey)
```

To deploy a new DataUnion, call:

```
dataUnionClient.deployDataUnion(name, adminAddress, adminFeeFractionWei, agents)
```

To get an existing instance of Data Union, call:
```
dataUnionClient.dataUnionFromName(name)
```

File: README.md [hunk 4]
```diff
@@ -361,6 +362,70 @@ To stop receiving events from a stream, pass the `Subscription` object you got w
...
+### Functions
+
+| Name          | Returns  | Description |
+| :------------ | :------ | :----------- |
+| isDeployed() | boolean | Check if Data Union deployment has completed |
+| addMembers(String ...members) | Transaction receipt | Add members |
+| partMembers(String ...members) | Transaction receipt | Remove members from Data Union |
+| withdrawTokensForSelfOrAsAdmin(String memberAddress, BigInteger amount) | Transaction receipt | Withdraw members tokens to given address |
+| withdrawTokensForMember(BigInteger privateKey, String to, BigInteger amount) | Transaction receipt | Withdraw members tokens |
+
```
Comment (fonty1) [inline README.md:381]: + CreateSecret
https://github.com/streamr-dev/streamr-client-javascript#admin-functions

File: README.md [hunk 5]
```diff
@@ -361,6 +362,70 @@ To stop receiving events from a stream, pass the `Subscription` object you got w
...
+DataUnion dataUnion = dataUnionClient.deployDataUnion("Cool Data Union", adminAddress, adminFee, agents);
+```
+
+### Member functions
+
+| Name                                  | Returns  |
+| :------------------------------------ | :------- |
+| isMemberActive(String memberAddress)         | boolean  |
+| isMemberInactive(String memberAddress)       | boolean  |
+| withdrawAllTokensForMember(String from, String to, byte[] signedWithdrawalRequest) | Transaction receipt |
```
Comment (fonty1) [inline README.md:397]: `Join` and possibly the sign withdrawal methods should be added
https://github.com/streamr-dev/streamr-client-javascript#member-functions

**Review by isentropy — APPROVED** [review]

**Pull request merged**